crates/mfem/examples/ex1.rs

```diff
@@ -133,24 +116,44 @@ fn main() -> anyhow::Result<()> {
     let mut a_mat = OperatorHandle::new();
     let mut b_vec = Vector::new();
     let mut x_vec = Vector::new();
-    a.form_linear_system(&ess_tdof_list, &x, &b, &mut a_mat, &mut x_vec, &mut b_vec);
-
-    println!("Size of linear system: {}", a_mat.height());
+    a.form_linear_system(
+        &ess_tdof_list,
+        &mut x,
+        &mut b,
```

Comment:
Having all these mutable is kind of an eye sore, but I can't think of a good way to fix that without either
- potentially breaking Rust memory model invariants
- making changes to MFEM upstream

I'm okay with reflecting the mutability semantics of the upstream MFEM C++ API. That will help us better understand what kind of changes to suggest there.

Reply:
I agree.  In a previous attempt I changed the interface slightly to be more “ergonomic” https://github.com/Chris00/mfem-rust/blob/main/crates/mfem/examples/ex1.rs#L123.  Another such attempt may be useful in the future.